src/internal.c

```diff
@@ -5024,7 +5024,7 @@ int EccSharedSecret(WOLFSSL* ssl, ecc_key* priv_key, ecc_key* pub_key,
 
 #ifdef WOLFSSL_ASYNC_CRYPT
     /* initialize event */
-    if (priv_key != NULL) {
+    if (!nonblockInProg && priv_key != NULL) {
```

Comment:
Could you eliminate the `nonblockInProg` and do the required setup for ECC inside the async.c code?